src/components/cards/WalletRestoreCard.tsx

```diff
@@ -23,18 +24,18 @@ interface Props {
  */
 export const WalletRestoreCard = (props: Props) => {
   const { walletInfo, onPress } = props
-  const { type, created } = walletInfo
 
   const theme = useTheme()
   const styles = getStyles(theme)
 
-  const pluginId = type.replace('wallet:', '')
-  const currencyConfig = useSelector(state => state.core.account.currencyConfig[pluginId])
-  const { currencyCode, displayName } = currencyConfig?.currencyInfo ?? {}
+  const account = useSelector(state => state.core.account)
+  const pluginId = Object.keys(account.currencyConfig).find(pluginId => account.currencyConfig[pluginId].currencyInfo.walletType === walletInfo.type)
+  const currencyConfig = pluginId == null ? undefined : account.currencyConfig[pluginId]
+  const { currencyCode, displayName } = findCurrencyInfo(account, walletInfo.type) ?? {}
```

Comment:
Oh, the `findCurrencyInfo` helper is pretty nice. We can use it to get the `pluginId` too:

```ts
const account = useSelector(state => state.core.account)
const { currencyCode, displayName, pluginId } = findCurrencyInfo(account, walletInfo.type) ?? {}
const currencyConfig = pluginId == null ? undefined : account.currencyConfig[pluginId]
```